File: gex/src/model/BarMatchTeamDeath.ts

export class BarMatchTeamDeath {
    public gameID: string = "";
    public teamID: number = 0;
    public reason: number = 0; // 2 = resign, 4 = killed
    public gameTime: number = 0;

    public static parse(elem: any): BarMatchTeamDeath {
        return {
            ...elem
        };
    }
}

File: gex/Models/Db/BarMatchTeamDeath.cs
using Dapper.ColumnMapper;
using gex.Code;

namespace gex.Models.Db {

    [DapperColumnsMapped]
    public class BarMatchTeamDeath {

        [ColumnMapping("game_id")]
        public string GameID { get; set; } = "";

        [ColumnMapping("team_id")]
        public byte TeamID { get; set; }

        [ColumnMapping("reason")]
        public byte Reason { get; set; }

        [ColumnMapping("game_time")]
        public float GameTime { get; set; }

    }
}
